bql/semantic/hooks.go

```diff
@@ -706,9 +703,38 @@ func addBindingToWorkingFilter(bndg string, workingFilter *FilterClause) error {
 	return nil
 }
 
-// isValidFilterClause returns true if the given filter clause is valid and complete.
-func isValidFilterClause(f *FilterClause) bool {
-	return f != nil && !f.Operation.IsEmpty() && f.Binding != ""
+// addValueToWorkingFilter takes the given value and tries to add it to workingFilter.
+func addValueToWorkingFilter(value string, workingFilter *FilterClause) error {
+	if workingFilter == nil {
+		return fmt.Errorf("could not add value %q to nil filter clause (which is still nil probably because a call to st.ResetWorkingFilterClause was not made before start processing the first filter clause)", value)
+	}
+	if workingFilter.Value != "" {
+		return fmt.Errorf("invalid value %q on filter clause since already set to %q", value, workingFilter.Value)
+	}
+
+	workingFilter.Value = value
+	return nil
+}
+
+// validateFilterClause returns an error if the given filter clause is either invalid or incomplete.
+func validateFilterClause(f *FilterClause) error {
+	if f == nil {
+		return fmt.Errorf("nil filter clause")
+	}
+	if f.Operation.IsEmpty() {
+		return fmt.Errorf("filter clause Operation is missing")
+	}
+	if f.Binding == "" {
+		return fmt.Errorf("filter clause Binding is missing")
+	}
+	if f.Value == "" && filter.OperationRequiresValue[f.Operation] {
+		return fmt.Errorf("filter clause Value is required for filter Operation %q", f.Operation)
+	}
+	if f.Value != "" && !filter.OperationRequiresValue[f.Operation] {
+		return fmt.Errorf("filter clause Value is inconsistent with filter Operation %q", f.Operation)
```

Comment:
I guess we can make this error more clear.
What about something like:
`filter clause Value is not required for filter Operation %q`

Or something in these lines?

Reply:
When writing this error message I was in doubt between "inconsistent" and "not required" too. Thank you for the comment Thiago, I have just proceeded with "not required" then. : )